sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/TryStatementsWithIdenticalCatchShouldBeMerged.cs

```diff
@@ -0,0 +1,141 @@
+            finally { }
+
+            try { } // Noncompliant {{Combine this 'try' with the one starting on line 9.}}
+            catch (Exception)
+            {
+            }
+            finally { }
+
+            try { }
+            finally { }
+
+            try { } // Noncompliant {{Combine this 'try' with the one starting on line 15.}}
+            catch (Exception)
+            {
+            }
+            finally { }
+
+            try { } // Noncompliant {{Combine this 'try' with the one starting on line 21.}}
+            finally { }
+        }
+
+        void DifferentCatches()
+        {
+            try { }
+            catch (Exception)
+            {
+            }
+
+            // exception type different
+            try { }
+            catch (ApplicationException)
+            {
+            }
+
+            // catch clauses count different
+            try { }
+            catch (Exception)
+            {
+            }
+            catch (ApplicationException)
+            {
+            }
+
+            // catch content different
+            try { }
+            catch (Exception)
+            {
+                Console.WriteLine();
+            }
+
+            // has finally
+            try { }
+            catch (Exception)
+            {
+            }
+            finally { }
+
+            // differs than previous by finally content
+            try { }
+            catch (Exception)
+            {
+            }
+            finally
+            {
+                Console.WriteLine();
+            }
+        }
+
+        void DifferentCatches()
+        {
+            try { }
+            catch (ApplicationException)
+            {
+            }
+            catch (Exception)
+            {
+            }
+
+            try { } // Noncompliant, same catches, different order
+            catch (Exception)
+            {
+            }
+            catch (ApplicationException)
+            {
+            }
+        }
+
+        void TryStatementsDifferentNesting()
+        {
+            try
+            {
+                // Child, not on the same level
+                try { }
+                catch (Exception)
+                {
+                }
+                catch (ApplicationException)
+                {
+                }
+            }
+            catch (ApplicationException)
+            {
+            }
+            catch (Exception)
+            {
+            }
+
+            if (true)
+            {
+                // Not on the same level
+                try { }
+                catch (Exception)
+                {
+                }
+                catch (ApplicationException)
+                {
+                }
+            }
+
+            try { } // Noncompliant {{Combine this 'try' with the one starting on line 102.}}
+            catch (Exception)
+            {
+            }
+            catch (ApplicationException)
+            {
+            }
+        }
+    }
+}
```

Comment:
Some other test cases:
* should this method work for properties too?
* non-compliant examples where the identical catch blocks contain multiple lines
* non-compliant examples where the identical try blocks contain multiple lines
* lambdas containing try-catch blocks that could be merged
* lambdas containing try-catch blocks in between outer blocks that could be merged
* ?nested functions - can we test for those?

Reply:
- done
- not really relevant imo, changed the non-compliant examples from 1 to 2 lines
- we don't really care about the try block, why write tests?
- done
- [ ] what does this mean?
- this should work
- Unfortunately the `SynaxNode.IsEquivalentTo` considers catch clauses with different variable names of the same exception type as different. E.g. `catch(Exception e)` and `catch(Exception ex)` is a FN.
- done

Reply:
2: my mistake, I think I missed the test case that with a non-empty catch block.
3: try blocks - good point, I was forgetting we didn't care about the content.
4: _lambdas containing..._ I meant a try block containing a lamba with an equivalent try block. But I guess the existing nesting tests are enough.
6: _nested functions ... this should work_ :  I guess this is just another type of nesting.